### PR TITLE
interactively query for license, suggest SPDX licenses

### PR DESCRIPTION
**Pull request by WebFreak001**
- add select input (also for sdl/json)
- changed to SPDX licensing (#2403)
- roughly sorted by popularity (kept similar licenses together though)

This is a rebase of #1101

### CODE REVIEW

**Comment by WebFreak001** [comment]
updated, now allows entering raw values if they are present in the list (e.g. sdl, json; for backwards compatibility) and displays items in column order, not in row order now. (User feedback with sample size of 2 preferred this ordering)

Example output:
```
Select a package recipe format
1) sdl    2) json   
#? [2]: 
Name [a]: 
Description [A minimal D application.]: 
Author name [webfreak]: 
Select or enter a license (https://choosealicense.com/)
 1) MIT                   5) BSD 3-clause          9) Apache-2.0           
 2) BSL-1.0 (Boost)       6) BSD 2-clause         10) MPL-2.0 (Mozilla)    
 3) public domain         7) GPL-2.0              11) AGPL-3.0             
 4) GPL-3.0               8) LGPL-3.0             12) proprietary          
? [12]: 
Copyright string [Copyright © 2023, webfreak]: 
Add dependency (leave empty to skip) []: 
     Success created empty project in /home/webfreak/dev/dub/a
             Package successfully created in .
```

File: source/dub/commandline.d [hunk 1]
```diff
@@ -997,7 +1048,13 @@ class InitCommand : Command {
 			}
 			p.description = input("Description", p.description);
 			p.authors = input("Author name", author).split(",").map!(a => a.strip).array;
-			p.license = input("License", p.license);
+			p.license = select(
+				"a license (https://choosealicense.com/)",
+				free_choice,
+				p.license,
+				"MIT", "BSL-1.0 (Boost)", "public domain", "GPL-3.0", "BSD 3-clause", "BSD 2-clause",
+				"GPL-2.0", "LGPL-3.0", "Apache-2.0", "MPL-2.0 (Mozilla)", "AGPL-3.0", "proprietary"
```
Comment (rikkimax) [inline source/dub/commandline.d:1056]: Artistic-v2 could be added (I use it).

Also, the CC's although there should be a warning asking are you sure?

**Comment by adamdruppe** [comment]
lgtm i think

File: source/dub/commandline.d (hunk 1 above)
Comment (WebFreak001) [reply source/dub/commandline.d:1056]: I think I used the license list from the PR I rebased here, it probably originates from https://dub.pm/package-format-json.html#licenses, although there are a few more there.

**Comment by WebFreak001** [comment]
just saw the license list is actually problematic, will adjust it to use what's documented on dub right now. We can change it later for properly specified licenses, e.g. #2403

**Comment by WebFreak001** [comment]
preview:
image
image

**Comment by WebFreak001** [comment]
aggregated nearly all the licenses used on DUB:

```
849    MIT
628    BSL-1.0
120    Apache-2.0
96    LGPL-3.0
67    GPL-3.0
61    public domain
61    proprietary
58    BSD 3-clause
42    Zlib
38    BSD 2-clause
26    MPL-2.0
20    GPL-2.0
20    CC0
19    ISC
17    BSD
16    AGPL-3.0
14    LGPL-2.1
12    LGPL-1.0
12    GPL-3.0 or later
9    GPL-1.0
8    ESL-1.0
7    WTFPL
7    Unlicense
7    LGPL-2.1 or later
7    GPL-2.0 or later
4    MS-PL
4    Apache-1.0
3    OpenSSL or SSLeay
3    NYSL
3    LGPL-3.0-or-later
2    LGPL 2.0+
2    FreeBSD
2    Fair Use License v2
2    CC Attribution-ShareAlike 
2    ASL-2.0
2    AGPL-3.0 or later
1    X11
1    WTFPL-2.0
1    VST
1    utf8proc
1    UIM
1    PSFv2
1    OpenLDAP
1    NIT
1    NetCDF
1    Microline
1    LGPL license with additional exceptions
1    LGPL 2.0
1    FreeBSD 2-clause
1    FOSS
1    EPL-1.0
1    DarkRiDDeR
1    CDDL
1    BSD-like(2-clause)
1    BSD-like
1    BSD 2-Clause "Simplified" License
1    Anti-996
1    ALGP3
```

**Comment by WebFreak001** [comment]
@Geod24 what do you think of this one to start the transition to SPDX?

Next I would change the dub docs to suggest using SPDX licenses and then inside DUB add an SPDX checker and emit warnings when the used license is invalid.

File: source/dub/commandline.d (hunk 1 above)
Comment (WebFreak001) [reply source/dub/commandline.d]: I have sorted by usage popularity now, didn't actually see Artistic-v2 used in any published DUB projects yet, do you have one?

You can also still manually type in text, so you can manually type Artistic-v2

**Pull request merged**